azext_iot/operations/events3/_parser.py

```diff
@@ -240,20 +270,50 @@ def _parse_payload(
 
         return payload
 
-    def _parse_annotations(self, message: Message):
+    def _validate_payload_against_dcm(
+        self,
+        origin_device_id: str,
+        payload: str,
+        central_device_provider: CentralDeviceProvider,
+        create_payload_name_error=False,
+    ):
+        if not central_device_provider:
+            return
+
+        if not hasattr(payload, "keys"):
+            # some error happend while parsing
+            # should be captured by _parse_payload method above
+            return
+
         try:
-            return unicode_binary_map(message.annotations)
-        except Exception:
-            self._warnings.append(
-                "Unable to decode message.annotations: {}".format(message.annotations)
+            template = central_device_provider.get_device_template(origin_device_id)
+        except Exception as e:
+            self._errors.append(
+                "Unable to get DCM for device: {}."
+                "Inner exception: {}".format(origin_device_id, e)
             )
+            return
 
-    def _parse_application_properties(self, message: Message):
-        try:
-            return unicode_binary_map(message.application_properties)
-        except Exception:
-            self._warnings.append(
-                "Unable to decode message.application_properties: {}".format(
-                    message.application_properties
+        all_schema = self._extract_schema_from_template(template)
+        all_names = [schema["name"] for schema in all_schema]
+
+        for telemetry_name in payload.keys():
+            if create_payload_name_error or telemetry_name not in all_names:
+                self._errors.append(
+                    "Telemetry item '{}' is not present in DCM. "
+                    "Device ID: {}. "
+                    "List of allowed telemetry values for this type of device: {}. "
+                    "NOTE: telemetry names are CASE-SENSITIVE".format(
+                        telemetry_name, origin_device_id, all_names
+                    )
                 )
-            )
+
+    def _extract_schema_from_template(self, template):
+        all_schema = []
+        dcm = template["capabilityModel"]
```

Comment:
Assuming there's no chance the template format is off. We need to be careful about crashing the monitor from malformed data.